project_euler/problem_04/sol1.py

```diff
@@ -20,23 +24,28 @@ def solution(n):
     39893
     """
     # fetches the next number
-    for number in range(n - 1, 10000, -1):
+    for number in range(n - 1, 9999, -1):
 
         # converts number into string.
-        strNumber = str(number)
+        str_number = str(number)
 
-        # checks whether 'strNumber' is a palindrome.
-        if strNumber == strNumber[::-1]:
+        # checks whether 'str_number' is a palindrome.
+        if str_number == str_number[::-1]:
 
             divisor = 999
 
             # if 'number' is a product of two 3-digit numbers
             # then number is the answer otherwise fetch next number.
             while divisor != 99:
-                if (number % divisor == 0) and (len(str(int(number / divisor))) == 3):
+                if (number % divisor == 0) and (len(str(number // divisor)) == 3):
                     return number
                 divisor -= 1
+    raise ValueError("No such latest palindrome number.")
```

Comment:
This error message is incorrect.  It is not the the number does not exist, it is that it is larger than our acceptable range.

Reply:
Can you give an error message ？ Thanks